apps/odyssey-stats/src/app.tsx

```diff
@@ -61,6 +61,28 @@ async function AppBoot() {
 	setStore( store as Store & WithAddReducer );
 	setupContextMiddleware( store, queryClient );
 
+	// Initialize site data early in the app boot process.
+	await new Promise< void >( ( resolve ) => {
```

Comment:
We have full control of the siteSelection function, and I guess we could just move the functionality here, and remove it in routes, which would be a little cleaner I guess?